internal/service/sagemaker/project_test.go

```diff
@@ -0,0 +1,398 @@
+package sagemaker_test
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/sagemaker"
+	sdkacctest "github.com/hashicorp/terraform-plugin-sdk/v2/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+	"github.com/hashicorp/terraform-provider-aws/internal/acctest"
+	"github.com/hashicorp/terraform-provider-aws/internal/conns"
+	tfsagemaker "github.com/hashicorp/terraform-provider-aws/internal/service/sagemaker"
+	"github.com/hashicorp/terraform-provider-aws/internal/tfresource"
+)
+
+func TestAccSageMakerProject_basic(t *testing.T) {
+	var mpg sagemaker.DescribeProjectOutput
+	rName := sdkacctest.RandomWithPrefix(acctest.ResourcePrefix)
+	resourceName := "aws_sagemaker_project.test"
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { acctest.PreCheck(t) },
+		ErrorCheck:   acctest.ErrorCheck(t, sagemaker.EndpointsID),
+		Providers:    acctest.Providers,
+		CheckDestroy: testAccCheckProjectDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccProjectBasicConfig(rName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckProjectExists(resourceName, &mpg),
+					resource.TestCheckResourceAttr(resourceName, "project_name", rName),
+					acctest.CheckResourceAttrRegionalARN(resourceName, "arn", "sagemaker", fmt.Sprintf("project/%s", rName)),
+					resource.TestCheckResourceAttr(resourceName, "service_catalog_provisioning_details.#", "1"),
+					resource.TestCheckResourceAttrPair(resourceName, "service_catalog_provisioning_details.0.product_id", "aws_servicecatalog_product.test", "id"),
+					resource.TestCheckResourceAttr(resourceName, "tags.%", "0"),
+					resource.TestCheckResourceAttrSet(resourceName, "project_id"),
+				),
+			},
+			{
+				ResourceName:      resourceName,
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+			{
```

Comment:
this is needed to successfully delete a project because of dependencies with iam roles. i couldnt crack the dependency graph.

example for failure on deletion of the whole state at once:

```
Client error: Termination failed with null: AmazonCloudFormationException  User: arn:aws:sts::****:assumed-role/tf-acc-test-****/servicecatalog is not authorized to perform: cloudformation:DeleteStack on resource: arn:aws:cloudformation:us-west-2:****:stack/SC-******/**** because no identity-based policy allows the cloudformation:DeleteStack action (Service: AmazonCloudFormation; Status Code: 403; Error Code: AccessDenied; Request ID: *****; Proxy: null)
```